### PR TITLE
Clean up unused mount points

### PR DESCRIPTION
**Pull request by derekbit**
Longhorn/longhorn#5391

### CODE REVIEW

File: util/util.go [hunk 1]
```diff
@@ -331,3 +335,33 @@ func MountWithTimeout(mounter mount.Interface, source string, target string, fst
...
+		if err != nil {
+			errs = multierr.Append(errs, errors.Wrapf(err, "failed to check if %s is mounted", path))
+			return nil
+		}
+
+		if mounted {
+			if err := cleanupMount(path, mounter, log); err != nil {
+				errs = multierr.Append(errs, errors.Wrapf(err, "failed to clean up mount point %v", path))
+			}
+		}
```
Comment (innobead) [inline util/util.go:362]: This part needs to update due to the `IsMountPoint` is incompatible with the version used by longhorn-manager.

File: util/util.go [hunk 2]
```diff
@@ -331,3 +335,33 @@ func MountWithTimeout(mounter mount.Interface, source string, target string, fst
 	}
 	return err
 }
+
+func CleanUpMountPoints(mounter mount.Interface, log logrus.FieldLogger) error {
```
Comment (innobead) [inline util/util.go:339]: Add a doc for this method.

**Review by innobead — APPROVED** [review]
In general, LGTM.

However, this means this library will only support one mount point at once. Suggest creating an improvement issue to make the library support multiple mounts capability for the future. (ex: support mutliple backups to different backup stores)

**Review by innobead — CHANGES_REQUESTED** [review]
Forgot some comments that need to resolve first.

File: util/util.go (hunk 2 above)
Comment (derekbit) [reply util/util.go:339]: Hmm, where is the doc?

Comment (innobead) [reply util/util.go:339]: Something like

```go
// CleanUpMountPoints will try to clean up all existing mount points for existing backup stores
func CleanUpMountPoints(mounter mount.Interface, log logrus.FieldLogger) error {
...
```

File: util/util.go (hunk 1 above)
Comment (derekbit) [reply util/util.go:374]: Add a feature request https://github.com/longhorn/longhorn/issues/5411.
Thank you.

File: util/util.go (hunk 2 above)
Comment (derekbit) [reply util/util.go:353]: Added. Thank you.

**Review by innobead — APPROVED** [review]
LGTM. :+1: ! 

Awesome to add docs for other exported functions :slightly_smiling_face:

**Pull request merged**